### PR TITLE
Add result flattening for the general result log. On by default.

### PR DESCRIPTION
**Pull request by Paul-Ferrell**

### CODE REVIEW

**Comment by Paul-Ferrell** [comment]
Closes #309

File: lib/pavilion/test_run.py [hunk 1]
```diff
@@ -781,15 +780,31 @@ def gather_results(self, run_result, regather=False, log_file=None):
...
+            base = results.copy()
+            del base['per_file']
+
+            for per_file, values in results['per_file'].items():
+                per_result = base.copy()
+                per_result['file'] = per_file
+                per_result.update(values)
+
+                result_logger.info(output.json_dumps(per_result))
+        else:
```
Comment (nicholas-sly) [inline lib/pavilion/test_run.py:805]: Is this strictly an either-or situation, or should this be done in both cases (minus the per_file content)?

File: test/tests/result_tests.py [hunk 2]
```diff
@@ -544,4 +544,43 @@ def test_re_search(self):
         for key, answer in answers.items():
             self.assertEqual(results[key], answer)
 
+    def test_flatten_results(self):
```
Comment (nicholas-sly) [inline test/tests/result_tests.py:547]: If this functionality should support updating per_file results as well as a global result, that should be tested here as well.

File: lib/pavilion/test_run.py (hunk 1 above)
Comment (Paul-Ferrell) [reply lib/pavilion/test_run.py:805]: If there's no per_file content, there's nothing to flatten.

File: test/tests/result_tests.py (hunk 2 above)
Comment (Paul-Ferrell) [reply test/tests/result_tests.py:547]: I extended the test.

**Pull request merged**